code/modules/research/server.dm

```diff
@@ -230,3 +241,145 @@
 	..()
 	playsound(src, "sparks", 75, 1)
 	to_chat(user, "<span class='notice'>You disable the security protocols.</span>")
+
+/// Master R&D server. As long as this still exists and still holds the HDD for the theft objective, research points generate at normal speed. Destroy it or an antag steals the HDD? Half research speed.
+/obj/machinery/rnd/server/master
+	var/obj/item/computer_hardware/hard_drive/cluster/hdd_theft/source_code_hdd
+	var/deconstruction_state = HDD_PANEL_CLOSED
+	var/front_panel_screws = 4
+	var/hdd_wires = 6
+
+/obj/machinery/rnd/server/master/Initialize(mapload)
+	. = ..()
+	name = "\improper Master " + name
+	source_code_hdd = new(src)
+	SSresearch.master_servers += src
+
+	add_overlay("RD-server-objective-stripes")
+
+/obj/machinery/rnd/server/master/Destroy()
+	if(source_code_hdd)
+		QDEL_NULL(source_code_hdd)
+
+	SSresearch.master_servers -= src
+
+	return ..()
+
+/obj/machinery/rnd/server/master/examine(mob/user)
+	. = ..()
+
+	switch(deconstruction_state)
+		if(HDD_PANEL_CLOSED)
+			. += "The front panel is closed. You can see some recesses which may have <b>screws</b>."
+		if(HDD_PANEL_OPEN)
+			. += "The front panel is dangling open. The hdd is in a secure housing. Looks like you'll have to <b>pry</b> it loose."
+		if(HDD_PRIED)
+			. += "The front panel is dangling open. The hdd has been pried from its housing. It is still connected by <b>wires</b>."
+		if(HDD_CUT_LOOSE)
+			. += "The front panel is dangling open. All you can see inside are cut wires and mangled metal."
+		if(HDD_OVERLOADED)
+			. += "The front panel is dangling open. The hdd inside is destroyed and the wires are all burned."
+
+/obj/machinery/rnd/server/master/tool_act(mob/living/user, obj/item/tool, tool_type)
+	// Only antags are given the training and knowledge to disassemble this thing.
+	if(is_special_character(user))
```

Comment:
just check if they have the objective.

Reply:
While minds have has_objective, steal objectives are all one objective, so it wont be able to know specifically the drive. Do you want this functionality added?

Reply:
you can check specific objective types, they don't all have the same type.

Reply:
If you check if `stealtype` (var on the steal objective) is `/datum/objective_item/steal/hdd_extraction` then you can confirm they have the objective.